### ISSUE TITLE
Failing instruction is not in RVM space

### ISSUE THREAD

**Issue by qinsoon** [mmtk/mmtk-jikesrvm#78]
It is a random bug. I think I have seen it a few times before. The most recent occurrence before this issue was the run for the PR: https://github.com/mmtk/mmtk-core/pull/381. But I don't think it is introduced by the PR.
```
./dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/rvm -X:gc:threads=16 -Xms75M -Xmx75M -jar benchmarks/dacapo-2006-10-MR2.jar lusearch
===== DaCapo lusearch starting =====
4 query batches completed
8 query batches completed
12 query batches completed
16 query batches completed
20 query batches completed
24 query batches completed
28 query batches completed
32 query batches completed
JikesRVM: Failing instruction starting at f7e22030 wasn't in RVM address space
00400000-00800000 rwxp 00000000 00:00 0 
JikesRVM: unexpected hardware trap outside of RVM address space - 0xf7e22030 0x300
2b000000-2b800000 rwxp 00000000 00:00 0 
fault address 0x3b7a0300
37800000-38000000 rwxp 00000000 00:00 0 
eip           0xf7e22030
3b400000-3c000000 rwxp 00000000 00:00 0 
eax (T0)      0x1b380000
3ec00000-3f000000 rwxp 00000000 00:00 0 
ebx (ctrs)    0xf7ef5eb0
56567000-5656c000 r-xp 00000000 08:01 1568092                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/JikesRVM
ecx (S0)      0xd9f0003c
5656c000-5656d000 r-xp 00004000 08:01 1568092                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/JikesRVM
edx (T1)      0xf7ef0101
5656d000-5656e000 rwxp 00005000 08:01 1568092                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/JikesRVM
esi (TR)      0x300
57878000-5908c000 rwxp 00000000 00:00 0                                  [heap]
edi (S1)      0x6957800c
60000000-62174000 rwxp 00000000 08:01 1568028                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/RVM.data.image
ebp           0x68d09ba8
64000000-6521a000 rwxp 00000000 08:01 1568027                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/RVM.code.image
esp (SP)      0x68d09ba4
67000000-67074000 r-xp 00000000 08:01 1568030                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/RVM.rmap.image
cs            0x23
68000000-69c00000 rwxp 00000000 00:00 0 
ds            0x2b
ba100000-ba1f5000 rwxp 00000000 00:00 0 
es            0x2b
ba1f5000-ba200000 ---p 00000000 00:00 0 
fs            (nil)
ba200000-ba235000 rwxp 00000000 00:00 0 
gs            0x63
ba235000-ba300000 ---p 00000000 00:00 0 
ss            0x2b
ba300000-ba3ff000 rwxp 00000000 00:00 0 
trapno        0x0000000e
ba3ff000-ba400000 ---p 00000000 00:00 0 
err           0x00000006
ba400000-ba500000 rwxp 00000000 00:00 0 
eflags        0x00010206
ba500000-ba5fd000 rwxp 00000000 00:00 0 
fpregs        0xf571ca90
ba5fd000-ba600000 ---p 00000000 00:00 0 
oldmask       0x00020000
ba600000-ba6fe000 rwxp 00000000 00:00 0 
cr2           0x3b7a0300
ba6fe000-ba700000 ---p 00000000 00:00 0 
attempting to dump proc map ...
ba700000-ba800000 rwxp 00000000 00:00 0 
ba800000-ba8fe000 rwxp 00000000 00:00 0 
ba8fe000-ba900000 ---p 00000000 00:00 0 
ba900000-baa00000 rwxp 00000000 00:00 0 
baa00000-bab00000 rwxp 00000000 00:00 0 
bab00000-bac00000 rwxp 00000000 00:00 0 
bac00000-bad00000 rwxp 00000000 00:00 0 
bad00000-bae00000 rwxp 00000000 00:00 0 
bae00000-baf00000 rwxp 00000000 00:00 0 
baf00000-bb000000 rwxp 00000000 00:00 0 
bb000000-bb100000 rwxp 00000000 00:00 0 
bb100000-bb200000 rwxp 00000000 00:00 0 
bb200000-bb300000 rwxp 00000000 00:00 0 
bb300000-bb400000 rwxp 00000000 00:00 0 
bb400000-bb500000 rwxp 00000000 00:00 0 
bb500000-bb600000 rwxp 00000000 00:00 0 
bb600000-bb700000 rwxp 00000000 00:00 0 
bb700000-bb800000 rwxp 00000000 00:00 0 
bb800000-bb900000 rwxp 00000000 00:00 0 
bb900000-bba00000 rwxp 00000000 00:00 0 
bba00000-bbb00000 rwxp 00000000 00:00 0 
bbb00000-bbc00000 rwxp 00000000 00:00 0 
bbc00000-bbd00000 rwxp 00000000 00:00 0 
bbd00000-bbe00000 rwxp 00000000 00:00 0 
bbe00000-bbf00000 rwxp 00000000 00:00 0 
bbf00000-bc000000 rwxp 00000000 00:00 0 
bc000000-bc100000 rwxp 00000000 00:00 0 
bc100000-bc300000 rwxp 00000000 00:00 0 
bc300000-bc400000 rwxp 00000000 00:00 0 
bc400000-bc500000 rwxp 00000000 00:00 0 
bc500000-bc600000 rwxp 00000000 00:00 0 
bc600000-bc700000 rwxp 00000000 00:00 0 
bc700000-bc800000 rwxp 00000000 00:00 0 
d08f4000-d08f5000 ---p 00000000 00:00 0 
d08f5000-d18f5000 rwxp 00000000 00:00 0 
d48f8000-d48f9000 ---p 00000000 00:00 0 
d48f9000-d58f9000 rwxp 00000000 00:00 0 
d9f00000-d9fb9000 rwxp 00000000 00:00 0 
d9fb9000-da000000 ---p 00000000 00:00 0 
da100000-da300000 rwxp 00000000 00:00 0 
da300000-da500000 rwxp 00000000 00:00 0 
da500000-da700000 rwxp 00000000 00:00 0 
da700000-da900000 rwxp 00000000 00:00 0 
da900000-dab00000 rwxp 00000000 00:00 0 
dab00000-dad00000 rwxp 00000000 00:00 0 
dad00000-daf00000 rwxp 00000000 00:00 0 
daf00000-db100000 rwxp 00000000 00:00 0 
db100000-db300000 rwxp 00000000 00:00 0 
db300000-db500000 rwxp 00000000 00:00 0 
db500000-db700000 rwxp 00000000 00:00 0 
db700000-db900000 rwxp 00000000 00:00 0 
db900000-dbb00000 rwxp 00000000 00:00 0 
dbb00000-dbd00000 rwxp 00000000 00:00 0 
dbd00000-dbf00000 rwxp 00000000 00:00 0 
dbf00000-dc100000 rwxp 00000000 00:00 0 
dc100000-dc300000 rwxp 00000000 00:00 0 
dc300000-dc500000 rwxp 00000000 00:00 0 
dc500000-dc6fd000 rwxp 00000000 00:00 0 
dc6fd000-dc700000 ---p 00000000 00:00 0 
dc700000-dc900000 rwxp 00000000 00:00 0 
dc900000-dcb00000 rwxp 00000000 00:00 0 
dcb00000-dcd00000 rwxp 00000000 00:00 0 
dcd00000-dcf00000 rwxp 00000000 00:00 0 
dcf00000-dd100000 rwxp 00000000 00:00 0 
dd100000-dd300000 rwxp 00000000 00:00 0 
dd300000-dd400000 rwxp 00000000 00:00 0 
dd4e8000-dd4e9000 ---p 00000000 00:00 0 
dd4e9000-de4e9000 rwxp 00000000 00:00 0 
de4e9000-de4ea000 ---p 00000000 00:00 0 
de4ea000-df4ea000 rwxp 00000000 00:00 0 
df4ea000-df4eb000 ---p 00000000 00:00 0 
df4eb000-e04eb000 rwxp 00000000 00:00 0 
e04eb000-e04ec000 ---p 00000000 00:00 0 
e04ec000-e14ec000 rwxp 00000000 00:00 0 
e14ec000-e14ed000 ---p 00000000 00:00 0 
e14ed000-e24ed000 rwxp 00000000 00:00 0 
e24ed000-e24ee000 ---p 00000000 00:00 0 
e24ee000-e34ee000 rwxp 00000000 00:00 0 
e34ee000-e34ef000 ---p 00000000 00:00 0 
e34ef000-e44ef000 rwxp 00000000 00:00 0 
e44ef000-e44f0000 ---p 00000000 00:00 0 
e44f0000-e54f0000 rwxp 00000000 00:00 0 
e54f0000-e54f1000 ---p 00000000 00:00 0 
e54f1000-e64f1000 rwxp 00000000 00:00 0 
e64f1000-e64f2000 ---p 00000000 00:00 0 
e64f2000-e74f2000 rwxp 00000000 00:00 0 
e74f2000-e74f3000 ---p 00000000 00:00 0 
e74f3000-e84f3000 rwxp 00000000 00:00 0 
e84f3000-e84f4000 ---p 00000000 00:00 0 
e84f4000-e94f4000 rwxp 00000000 00:00 0 
e94f4000-e94f5000 ---p 00000000 00:00 0 
e94f5000-ea4f5000 rwxp 00000000 00:00 0 
ea4f5000-ea4f6000 ---p 00000000 00:00 0 
ea4f6000-eb4f6000 rwxp 00000000 00:00 0 
eb4f6000-eb4f7000 ---p 00000000 00:00 0 
eb4f7000-ec4f7000 rwxp 00000000 00:00 0 
ec4f7000-ec4f8000 ---p 00000000 00:00 0 
ec4f8000-ed4f8000 rwxp 00000000 00:00 0 
ed4f8000-ed4f9000 ---p 00000000 00:00 0 
ed4f9000-ee4f9000 rwxp 00000000 00:00 0 
ee4f9000-ee4fa000 ---p 00000000 00:00 0 
ee4fa000-ef4fa000 rwxp 00000000 00:00 0 
ef4fa000-ef4fb000 ---p 00000000 00:00 0 
ef4fb000-f04fb000 rwxp 00000000 00:00 0 
f04fb000-f04fc000 ---p 00000000 00:00 0 
f04fc000-f14fc000 rwxp 00000000 00:00 0 
f14fc000-f14fd000 ---p 00000000 00:00 0 
f14fd000-f24fd000 rwxp 00000000 00:00 0 
f24fd000-f24fe000 ---p 00000000 00:00 0 
f24fe000-f34fe000 rwxp 00000000 00:00 0 
f34fe000-f34ff000 ---p 00000000 00:00 0 
f34ff000-f44ff000 rwxp 00000000 00:00 0 
f44ff000-f4500000 ---p 00000000 00:00 0 
f4500000-f5500000 rwxp 00000000 00:00 0 
f5500000-f5600000 rwxp 00000000 00:00 0 
f5600000-f5700000 rwxp 00000000 00:00 0 
f5700000-f5800000 rwxp 00000000 00:00 0 
f583d000-f584d000 r-xp 00000000 08:01 1568090                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjavanio.so
f584d000-f584e000 r-xp 0000f000 08:01 1568090                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjavanio.so
f584e000-f584f000 rwxp 00010000 08:01 1568090                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjavanio.so
f584f000-f5850000 rwxp 00000000 00:00 0 
f5850000-f586a000 r-xp 00000000 08:01 1568086                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjavalang.so
f586a000-f586b000 r-xp 00019000 08:01 1568086                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjavalang.so
f586b000-f586c000 rwxp 0001a000 08:01 1568086                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjavalang.so
f586c000-f586d000 rwxp 00000000 00:00 0 
f586d000-f586e000 ---p 00000000 00:00 0 
f586e000-f686e000 rwxp 00000000 00:00 0 
f686e000-f686f000 ---p 00000000 00:00 0 
f686f000-f7ab2000 rwxp 00000000 00:00 0 
f7ab2000-f7ab5000 r-xp 00000000 08:01 5677065                            /lib32/libdl-2.27.so
f7ab5000-f7ab6000 r-xp 00002000 08:01 5677065                            /lib32/libdl-2.27.so
f7ab6000-f7ab7000 rwxp 00003000 08:01 5677065                            /lib32/libdl-2.27.so
f7ab7000-f7b80000 r-xp 00000000 08:01 5677066                            /lib32/libm-2.27.so
f7b80000-f7b81000 r-xp 000c8000 08:01 5677066                            /lib32/libm-2.27.so
f7b81000-f7b82000 rwxp 000c9000 08:01 5677066                            /lib32/libm-2.27.so
f7b82000-f7b9d000 r-xp 00000000 08:01 5677076                            /lib32/libpthread-2.27.so
f7b9d000-f7b9e000 r-xp 0001a000 08:01 5677076                            /lib32/libpthread-2.27.so
f7b9e000-f7b9f000 rwxp 0001b000 08:01 5677076                            /lib32/libpthread-2.27.so
f7b9f000-f7ba1000 rwxp 00000000 00:00 0 
f7ba1000-f7ba9000 r-xp 00000000 08:01 5677078                            /lib32/librt-2.27.so
f7ba9000-f7baa000 r-xp 00007000 08:01 5677078                            /lib32/librt-2.27.so
f7baa000-f7bab000 rwxp 00008000 08:01 5677078                            /lib32/librt-2.27.so
f7bab000-f7bc7000 r-xp 00000000 08:01 1053000                            /usr/lib32/libgcc_s.so.1
f7bc7000-f7bc8000 r-xp 0001b000 08:01 1053000                            /usr/lib32/libgcc_s.so.1
f7bc8000-f7bc9000 rwxp 0001c000 08:01 1053000                            /usr/lib32/libgcc_s.so.1
f7bc9000-f7d9b000 r-xp 00000000 08:01 5677062                            /lib32/libc-2.27.so
f7d9b000-f7d9c000 ---p 001d2000 08:01 5677062                            /lib32/libc-2.27.so
f7d9c000-f7d9e000 r-xp 001d2000 08:01 5677062                            /lib32/libc-2.27.so
f7d9e000-f7d9f000 rwxp 001d4000 08:01 5677062                            /lib32/libc-2.27.so
f7d9f000-f7da2000 rwxp 00000000 00:00 0 
f7da5000-f7daf000 r-xp 00000000 08:01 1568085                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjavaio.so
f7daf000-f7db0000 r-xp 00009000 08:01 1568085                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjavaio.so
f7db0000-f7db1000 rwxp 0000a000 08:01 1568085                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjavaio.so
f7db1000-f7db2000 rwxp 00000000 00:00 0 
f7db2000-f7db3000 r-xp 00000000 08:01 1568094                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjvm_jni.so
f7db3000-f7db4000 r-xp 00000000 08:01 1568094                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjvm_jni.so
f7db4000-f7db5000 rwxp 00001000 08:01 1568094                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjvm_jni.so
f7db5000-f7eef000 r-xp 00000000 08:01 1568095                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libmmtk.so
f7eef000-f7ef6000 r-xp 00139000 08:01 1568095                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libmmtk.so
f7ef6000-f7ef7000 rwxp 00140000 08:01 1568095                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libmmtk.so
f7ef7000-f7ef8000 rwxp 00000000 00:00 0 
f7ef8000-f7f0e000 r-xp 00000000 08:01 1568093                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjvm.so
f7f0e000-f7f0f000 r-xp 00015000 08:01 1568093                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjvm.so
f7f0f000-f7f10000 rwxp 00016000 08:01 1568093                            /home/runner/work/mmtk-core/mmtk-core/mmtk-jikesrvm/repos/jikesrvm/dist/RFastAdaptiveMarkSweep_x86_64_m32-linux/libjvm.so
f7f10000-f7f12000 rwxp 00000000 00:00 0 
f7f12000-f7f15000 r--p 00000000 00:00 0                                  [vvar]
f7f15000-f7f16000 r-xp 00000000 00:00 0                                  [vdso]
f7f16000-f7f3c000 r-xp 00000000 08:01 5677058                            /lib32/ld-2.27.so
f7f3c000-f7f3d000 r-xp 00025000 08:01 5677058                            /lib32/ld-2.27.so
f7f3d000-f7f3e000 rwxp 00026000 08:01 5677058                            /lib32/ld-2.27.so
ff7eb000-ff80d000 rwxp 00000000 00:00 0                                  [stack]
```

**Comment by qinsoon** [comment]
I have seen this bug with https://github.com/mmtk/mmtk-core/pull/392/checks?check_run_id=3198338962 in https://github.com/mmtk/mmtk-core/pull/392.

**Comment by qinsoon** [comment]
I think this bug was introduced in https://github.com/mmtk/mmtk-core/commit/f1b3ce4cf67149f724430406eea49f7a3b4c7fb2 (the marksweep change). 

I ran 111f4b58ea5357b2d848b69b51ac1d2b527f3622 with https://github.com/mmtk/mmtk-core/commit/0875c9084fc11114f122f4f4d59d27f2ceeae07b (the commit before the marksweep change). There was no issue. 
But if I ran 111f4b58ea5357b2d848b69b51ac1d2b527f3622 with https://github.com/mmtk/mmtk-core/commit/f1b3ce4cf67149f724430406eea49f7a3b4c7fb2, this bug can be reproduced (transient bug).
